src/lib/cardiac_mechanics/BaseSolverNeoHooke.hpp

```diff
@@ -0,0 +1,370 @@
+#include <iostream>
+#include <map>
+#ifdef BUILD_TYPE_DEBUG
+#warning "Building in DEBUG mode"
+#include <string>
+#endif
+
+using namespace dealii;
+
+/**
+ * @class BaseSolverNeoHooke
+ * @brief Abstract class representing a base solver for non linear cardiac
+ * mechanics by using the neo Hooke constitutive law.
+ * @tparam dim The problem dimension
+ * @tparam Scalar The scalar type for the problem, by default a double
+ */
+template <int dim, typename Scalar = double> class BaseSolverNeoHooke {
+  /**
+   * Alias for the base preconditioner pointer
+   */
+  using Preconditioner = std::unique_ptr<TrilinosWrappers::PreconditionBase>;
+  /**
+   * Alias for the base linear solver pointer
+   */
+  using LinearSolver =
+      std::unique_ptr<SolverBase<TrilinosWrappers::MPI::Vector>>;
+  /**
+   * Sacado automatic differentiation type code from
+   */
+  static constexpr Differentiation::AD::NumberTypes ADTypeCode =
+      Differentiation::AD::NumberTypes::sacado_dfad;
+  /**
+   * Alias for the AD helper
+   */
+  using ADHelper =
+      Differentiation::AD::ResidualLinearization<ADTypeCode, Scalar>;
+  /**
+   * Alias for the AD number type
+   */
+  using ADNumberType = typename ADHelper::ad_type;
+
+public:
+  /**
+   * @brief: Triangulation geometry
+   */
+  enum class TriangulationType { T, Q };
+
+  /**
+   * @brief Material related parameters.
+   */
+  struct Material {
+    /**
+     * mu coefficient
+     */
+    static Scalar mu;
+    /**
+     * lambda coefficient
+     */
+    static Scalar lambda;
+
+    /**
+     * Retrive a std::string representing the current configuration
+     * @return The configuration string
+     */
+    static std::string show() {
+      return "  mu: " + std::to_string(mu) + "\n" +
+             "  lambda: " + std::to_string(lambda) + "\n";
+    }
+  };
+  /**
+   * @brief Class representing the applied pressure component
+   */
+  class ConstantPressureFunction : Function<dim> {
+  public:
+    /**
+     * @brief Default constructor
+     */
+    ConstantPressureFunction() {}
+    /**
+     * @brief Parametrized constructor
+     * @param pressure_ The pressure value;
+     * @param reduction_factor_ The pressure reduction factor to accomodate the
+     * modified Newton iteration
+     * @param reduction_factor_inc_ The incremental value for the above
+     */
+    ConstantPressureFunction(const Scalar pressure_,
+                             const Scalar reduction_factor_,
+                             const Scalar reduction_factor_inc_)
+        : pressure(pressure_), reduction_factor(reduction_factor_),
+          reduction_increment_strat(reduction_factor_inc_) {}
+    /**
+     * @brief Copy operator
+     * @param other The input ConstantPressureFunction object
+     */
+    void operator=(ConstantPressureFunction &&other) {
+      pressure = other.pressure;
+      reduction_factor = other.reduction_factor;
+      reduction_increment_strat = other.reduction_increment_strat;
+    }
```

Comment:
I believe this operator would be default-generated, there should be no need to declare it manually. If you want to be explicit, you could say
```suggestion
    void operator=(ConstantPressureFunction &&other) = default;
```